tests/lib/rules/jsx-no-leaked-zero.js

```diff
@@ -0,0 +1,313 @@
+/**
+ * @fileoverview Prevent zero from being rendered on numerical condition
+ * @author Mario Beltrán
+ */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/jsx-no-leaked-zero');
+
+const parsers = require('../../helpers/parsers');
+
+const parserOptions = {
+  ecmaVersion: 2018,
+  sourceType: 'module',
+  ecmaFeatures: {
+    jsx: true,
+  },
+};
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions });
+ruleTester.run('jsx-no-leaked-zero', rule, {
+  valid: parsers.all([
+    {
+      code: `
+      const Component = () => {
+        return <div>{customTitle || defaultTitle}</div>
+      }
```

Comment:
```suggestion
        const Component = () => {
          return <div>{customTitle || defaultTitle}</div>
        }
```
(same for all the test cases)